test/build/55_cache.bats

```diff
@@ -468,6 +468,33 @@ EOF
 }
 
 
+@test "${tag}: branch delete" {
+    ch-image build-cache --reset
+
+    # Build image
+    ch-image build -t a -f bucache/a.df ./bucache
+    run ch-image build-cache --tree
+    echo "$output"
+    [[ $status -eq 0 ]]
+    diff -u - <(echo "$output" | treeonly) <<'EOF'
+*  (a) RUN echo bar
+*  RUN echo foo
+*  (alpine+3.9) PULL alpine:3.9
+*  (HEAD -> root) ROOT
+EOF
+
+    # Delete image
+    ch-image delete a
+    run ch-image build-cache --tree
+    echo "$output"
+    [[ $status -eq 0 ]]    
+    diff -u - <(echo "$output" | treeonly) <<'EOF'
+*  (alpine+3.9) PULL alpine:3.9
+*  (HEAD -> root) ROOT
```

Comment:
I'm surprised the branch is gone completely. Does it still hit if you re-build? I'd hope so. Probably ought to test that.